### PR TITLE
Bugfix/issue 285 cloudinit v2

### LINKED ISSUE #285: Cloud_init and sysprep CDROMs being detected and destroyed forcing VM reboot
### Terraform Version
- 0.12.17
- 0.13.7
- 0.15.3
- 1.0.0

### Affected Resource(s)
* nutanix_virtual_machine

### Terraform Configuration Files

```hcl
resource "nutanix_virtual_machine" "vm" {
  count = var.platform == "ahv" && local.os_short_name == "rhe" ? 1 : 0
  name  = infoblox_record_host.host.name
  cluster_uuid = element([for c in flatten(data.nutanix_clusters.clusters[*].entities) :
  c.metadata["uuid"] if c.name == var.cluster], 0)
  description          = var.owner != "" ? "${var.owner} - terraform - ${lookup(data.external.environment.result, "date", formatdate("YYYYMMDD", timestamp()))}" : "${lookup(data.external.environment.result, "owner", "terraform")} - terraform - ${lookup(data.external.environment.result, "date", formatdate("YYYYMMDD", timestamp()))}"
  num_vcpus_per_socket = var.cpus <= local.cpu_max ? var.cpus : local.cpu_max
  num_sockets          = 1
  memory_size_mib      = var.memory <= local.memory_max ? var.memory : local.memory_max

  owner_reference = {
    kind = "user"
    name = data.nutanix_user.user[count.index].user_name
    uuid = data.nutanix_user.user[count.index].id
  }

  nic_list {
    subnet_uuid = data.nutanix_subnet.subnet[count.index].id
  }

  guest_customization_cloud_init_user_data = base64encode(data.template_file.cloudinit_user_data[count.index].rendered)

  disk_list {
    data_source_reference = {
      kind = "image"
      uuid = data.nutanix_image.image[count.index].id
    }
    device_properties {
      device_type = "DISK"
      disk_address = {
        adapter_type = "SCSI"
        device_index = "0"
      }
    }
  }

  dynamic "disk_list" {
    iterator = iii
    for_each = var.additional_disks
    content {
      disk_size_mib = iii.value * 1000
    }
  }
  ... (provisioners only beyond this point)
}
```
### Actual Behavior
After initial apply to create the VM, a plan would display the following:
```hcl
      - disk_list {
          - data_source_reference  = {} -> null
          - disk_size_bytes        = 0 -> null
          - disk_size_mib          = 0 -> null
          - uuid                   = "77ebeb80-c890-4130-90fc-85c14d546819" -> null
          - volume_group_reference = {} -> null

          - device_properties {
              - device_type  = "CDROM" -> null
              - disk_address = {
                  - "adapter_type" = "IDE"
                  - "device_index" = "0"
                } -> null
            }
        }
```

### Steps to Reproduce
1. Create a `nutanix_virtual_machine` resource *without* specifying the creation of a CDROM, but specifying `guest_customization_cloud_init_user_data` or `guest_customization_sysprep`
2. After creation, run `terraform plan` again
3. Observe a CDROM is present, which Terraform wants to destroy
4. Run `terraform apply`, and observe the VM is rebooted

### Important Factors
This has happened using multiple different versions of Terraform, and with provider version 1.1.1 and 1.2.0, all against AOS 5.19.*. This is not present in 5.15 LTS, and we are yet to test 5.20 LTS.
​

### PR DESCRIPTION
**Pull request by yannickstruyf3**
fix #285

### CODE REVIEW

File: nutanix/structure.go [hunk 1]
```diff
@@ -177,73 +221,91 @@ func flattenDiskListFilterCloudInit(d *schema.ResourceData, disks []*v3.VMDisk)
...
+					if !removeCloudInit {
+						filteredDiskList = append(filteredDiskList, eDisk)
+					}
 				}
 			}
 		}
-		if len(potentialCloudInitIDs) == 1 {
-			cloudInitCdromUUID = potentialCloudInitIDs[0]
-			d.Set("cloud_init_cdrom_uuid", cloudInitCdromUUID)
+		log.Printf("[YST] pre if potentialCloudInitHelpers ")
```
Comment (coderGo93) [inline nutanix/structure.go]: is this log necessary?

File: nutanix/structure.go [hunk 2]
```diff
@@ -177,73 +221,91 @@ func flattenDiskListFilterCloudInit(d *schema.ResourceData, disks []*v3.VMDisk)
...
+		if len(potentialCloudInitHelpers) == 1 {
+			log.Printf("[YST] in if potentialCloudInitHelpers ")
```
Comment (coderGo93) [inline nutanix/structure.go]: same as above

**Review by coderGo93 — COMMENTED** [review]
It seems good but it has some logging line for debugging purposes but not sure if it's needed or not

File: nutanix/structure.go (hunk 1 above)
Comment (yannickstruyf3) [reply nutanix/structure.go]: Good catch. Removed log

File: nutanix/structure.go (hunk 2 above)
Comment (yannickstruyf3) [reply nutanix/structure.go]: Good catch. Removed log

File: nutanix/structure.go [hunk 3]
```diff
@@ -177,73 +220,87 @@ func flattenDiskListFilterCloudInit(d *schema.ResourceData, disks []*v3.VMDisk)
...
 				}
 			}
 		}
-		if len(potentialCloudInitIDs) == 1 {
-			cloudInitCdromUUID = potentialCloudInitIDs[0]
-			d.Set("cloud_init_cdrom_uuid", cloudInitCdromUUID)
+		if len(potentialCloudInitHelpers) == 1 {
+			cloudInitHelper = &potentialCloudInitHelpers[0]
+			cloudInitCdromUUID = cloudInitHelper.UUID
+			// d.Set("cloud_init_cdrom_uuid", cloudInitCdromUUID)
```
Comment (coderGo93) [inline nutanix/structure.go]: is it necessary to leave it as a comment?

Comment (yannickstruyf3) [reply nutanix/structure.go]: Removed.

**Review by coderGo93 — APPROVED** [review]
LGTM if the tests pass

**Review by marinsalinas — APPROVED** [review]

**Pull request merged**